flycheck.el

```diff
@@ -6920,6 +6921,25 @@ See URL `http://www.erlang.org/'."
    (error line-start (file-name) ":" line ": " (message) line-end))
   :modes erlang-mode)
 
+
+(flycheck-define-checker erlang-rebar3
+  "An Erlang syntax checker using the rebar3 build tool.
+
+See URL `https://www.rebar3.org'."
+  :command ("rebar3"
+            "compile")
+  :error-patterns
+  ((warning line-start
+            (file-name) ":" line ": Warning:" (message) line-end)
+   (error line-start
+          (file-name) ":" line ": " (message) line-end))
+  :modes erlang-mode
+  :working-directory flycheck-rebar3-project-root)
+
+(defun flycheck-rebar3-project-root (_checker)
+  "Return directory where =rebar.config= is located."
+  (locate-dominating-file buffer-file-name "rebar.config"))
```

Comment:
Does `rebar` work if `rebar.config` is missing?  If not it needs an `:enabled` function that checks whether the file is actually present.